src/node/node_class.hpp

```diff
@@ -1093,118 +1082,80 @@ inline std::vector<Napi::PropertyDescriptor> ObjectWrap<ClassType>::create_napi_
 			node::String* name = get_cached_property_name(propName);
 			auto descriptor = Napi::PropertyDescriptor::Accessor<property_getter, property_setter>(name->ToString(env), napi_enumerable, (void*)name);
 			properties.push_back(descriptor);
+			// XXX should computed properties be reported as enumerable own keys?
```

Comment:
This is a question for the team. I don't know which properties make sense to show up in a spread / have `instance.hasOwnProperty()` return true for.

If the model is that computed properties are like getters on a JS class, then they probably shouldn't. But if they are "like stored properties but computed on demand", then they probably should.

On the other hand, maybe this is something that calls for practicality over purity, and if it is more convenient for users if it shows up in a spread, maybe it is worth just making that happen, regardless of what the model would suggest. We should also be aware that it may be expensive to eagerly compute properties that are not needed, but I suppose that applies as much or more to fetching the fields from the DB.